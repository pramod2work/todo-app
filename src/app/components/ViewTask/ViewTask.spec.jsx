/* global describe test expect beforeAll jest */

import React from 'react'
import { shallow } from 'enzyme'

import ViewTasksComponent from './ViewTask'

describe('Test for View Tasks Component', () => {
  let component
  const propValues = {
    showList: true,
    tasks: [{
      name: 'test',
      completed: false,
      highPriority: false,
      id: 0,
      key: 0
    }],
    completedCount: 0,
    deleteTaskById: jest.fn(),
    updateTaskById: jest.fn(),
    history: { push: jest.fn() }
  }

  beforeAll(() => {
    component = shallow(<ViewTasksComponent {...propValues} />)
  })

  test('Check snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  test('Add Action', () => {
    component.find('[data-test-id="add-action"]').simulate('click')
    expect(propValues.history.push).toHaveBeenCalledWith('/task/add')
  })

  test('Update Action', () => {
    component.find('[data-test-id="update-action-0"]').simulate('click')
    expect(propValues.history.push).toHaveBeenCalledWith('/task/0')
  })

  test('Toggle Priority', () => {
    component.find('[data-test-id="update-priority-0"]').simulate('change')
    expect(propValues.updateTaskById).toHaveBeenCalledWith({
      ...propValues.tasks[0],
      highPriority: true
    })
  })

  test('Toggle Complete Status', () => {
    component.find('[data-test-id="update-completed-0"]').simulate('change')
    expect(propValues.updateTaskById).toHaveBeenCalledWith({
      ...propValues.tasks[0],
      completed: true
    })
  })

  test('Test Delete Action', () => {
    component.find('[data-test-id="delete-action-0"]').simulate('click')
    expect(propValues.deleteTaskById).toHaveBeenCalledWith(0)
  })
})
