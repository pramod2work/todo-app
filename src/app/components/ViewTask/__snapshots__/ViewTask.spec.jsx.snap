// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for View Tasks Component Check snapshot 1`] = `
<div>
  <ForwardRef(Bootstrap(Row))>
    <ForwardRef(Bootstrap(ButtonToolbar))>
      <ForwardRef(Bootstrap(Button))
        data-test-id="add-action"
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add Task
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(ButtonToolbar))>
  </ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Row))
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <ForwardRef(Bootstrap(Table))
      bordered={true}
      hover={true}
      responsive={true}
      striped={true}
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th
            colSpan="2"
          >
            Task
          </th>
          <th>
            High Priority
          </th>
          <th>
            Completed
          </th>
          <th
            colSpan="1"
          />
        </tr>
      </thead>
      <tbody>
        <tr
          key="0test"
        >
          <td>
            1
          </td>
          <td
            colSpan="2"
          >
            <ForwardRef(Bootstrap(Button))
              data-test-id="update-action-0"
              onClick={[Function]}
              type="button"
              variant="link"
            >
              test
            </ForwardRef(Bootstrap(Button))>
          </td>
          <td
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <ForwardRef(Bootstrap(FormCheck))
              checked={false}
              data-test-id="update-priority-0"
              inline={true}
              name="highPriority"
              onChange={[Function]}
            />
          </td>
          <td
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <ForwardRef(Bootstrap(FormCheck))
              checked={false}
              data-test-id="update-completed-0"
              inline={true}
              name="completed"
              onChange={[Function]}
            />
          </td>
          <td>
            <ForwardRef(Bootstrap(Button))
              data-test-id="delete-action-0"
              onClick={[Function]}
              type="button"
              variant="danger"
            >
              Delete
            </ForwardRef(Bootstrap(Button))>
          </td>
        </tr>
      </tbody>
    </ForwardRef(Bootstrap(Table))>
  </ForwardRef(Bootstrap(Row))>
  <ForwardRef(Bootstrap(Row))
    style={
      Object {
        "marginTop": "1rem",
      }
    }
  >
    <ForwardRef(Bootstrap(ButtonToolbar))>
      <ForwardRef(Bootstrap(Button))
        style={
          Object {
            "marginRight": "1rem",
          }
        }
        variant="success"
      >
        Completed 
        <ForwardRef(Bootstrap(Badge))
          data-test-id="completed-count"
          pill={true}
          variant="light"
        >
          0
        </ForwardRef(Bootstrap(Badge))>
      </ForwardRef(Bootstrap(Button))>
      <ForwardRef(Bootstrap(Button))
        variant="info"
      >
        Total 
        <ForwardRef(Bootstrap(Badge))
          data-test-id="total-count"
          pill={true}
          variant="light"
        >
          1
        </ForwardRef(Bootstrap(Badge))>
      </ForwardRef(Bootstrap(Button))>
    </ForwardRef(Bootstrap(ButtonToolbar))>
  </ForwardRef(Bootstrap(Row))>
</div>
`;
